Update Readme.md (#10)
DevExpress-Examples/how-to-sort-a-aspxpivotgrid-by-clicking-on-a-field-value-e1439: Readme.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/E1439)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->
# Pivot Grid for Web Forms - How to Sort a PivotGrid by Clicking on a Field Value

> The PivotGrid allows to sort by any column. To do this, right click any field value and select an appropriate field to sort by. Sometimes it is possible to sort by any column by clicking on it. This example shows how to implement this behavior. 

Reviewer: ```suggestion
The PivotGrid allows you to sort by any column. To do this, right click any field value and select an appropriate field by which to sort. Sometimes it is possible to sort by any column by clicking on it. This example shows how to implement this behavior.
```

> Create a template for field values to place a hyperlink (or a button) rather than a plain text. Use the pivot grid's [FieldValueTemplate](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxPivotGrid.ASPxPivotGrid.FieldValueTemplate) property.

Reviewer: `Create a template for field values to place a hyperlink (or a button) rather than a plain text.`

1. What do you mean by "place" in this context? For example, do you mean "display"?

2. Please change "rather than a plain text" to "rather than plain text".

3. `Use the pivot grid's [FieldValueTemplate](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxPivotGrid.ASPxPivotGrid.FieldValueTemplate) property.`

Can we update this sentence to the following?

`To create a template, use the pivot grid's [FieldValueTemplate](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxPivotGrid.ASPxPivotGrid.FieldValueTemplate) property.`

Author: ```suggestion
Create a template for field values to display a hyperlink (or a button) instead of plain text. To create a template, use the pivot grid's [FieldValueTemplate](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxPivotGrid.ASPxPivotGrid.FieldValueTemplate) property.
```
"Display" seems to be more exact than "place" in this case.

```cs
ASPxPivotGrid1.FieldValueTemplate = new FieldValueTemplate(ASPxPivotGrid1);
```

> The *FieldValueTemplate* class should implement the [ITemplate](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.itemplate?view=netframework-4.8) interface. Replace standard contents of a field value in the *InstantiateIn* method (see [Default.aspx.cs](./CS/Default.aspx.cs#L82-L131)/[Default.aspx.vb](./VB/Default.aspx.vb#L81-L134)).

Reviewer: ```suggestion
The *FieldValueTemplate* class should implement the [ITemplate](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.itemplate?view=netframework-4.8) interface. Replace the standard content of a field value in the *InstantiateIn* method (see [Default.aspx.cs](./CS/Default.aspx.cs#L82-L131)/[Default.aspx.vb](./VB/Default.aspx.vb#L81-L134)).
```

> The *GetOnClickHandler* method returns a JavaScript code that will perform a ASPxPivotGrid's callback passing all required parameters to the server. On the server side, handle the *CustomCallback* event handler and apply sorting to it (see [Default.aspx.cs](./CS/Default.aspx.cs#L20-L71)/[Default.aspx.vb](./VB/Default.aspx.vb#L24-L70)).

Reviewer: ```suggestion
The *GetOnClickHandler* method returns JavaScript code that will perform an ASPxPivotGrid callback, passing all required parameters to the server. On the server side, handle the *CustomCallback* event handler and apply sorting to it (see [Default.aspx.cs](./CS/Default.aspx.cs#L20-L71)/[Default.aspx.vb](./VB/Default.aspx.vb#L24-L70)).
```
